dynawo/sources/Modeler/DataInterface/PowSyblIIDM/DYNDataInterfaceIIDM.cpp

```diff
@@ -1083,6 +1105,28 @@ DataInterfaceIIDM::getStaticParameterBoolValue(const std::string& staticID, cons
   return findComponent(staticID)->getStaticParameterValue<bool>(refOrigName);
 }
 
+void
+DataInterfaceIIDM::copy(const DataInterfaceIIDM& other) {
+  networkIIDM_  = other.networkIIDM_;  // No clone here because iidm network is not copyable
+  // Criterias are not copied and must be initialized again
+  serviceManager_ = boost::make_shared<ServiceManagerInterfaceIIDM>(this);
```

Comment:
Why do we need to copy the service manager?

Reply:
the service manager references the data interface IIDM instance